nmap_ftp.cc

```diff
@@ -167,9 +168,9 @@ int parse_bounce_argument(struct ftpinfo *ftp, char *url) {
     ftp->port = atoi(s);
   }
 
-  strncpy(ftp->server_name, q, MAXHOSTNAMELEN);
+  strncpy(ftp->server_name, q, FQDN_LEN);
```

Comment:
This is allocated in `nmap_ftp.h` as `char server_name[MAXHOSTNAMELEN + 1];`, so that needs to be changed, too. Specifically, try this formula to find all instances of MAXHOSTNAMELEN: `vim -q <(git grep -n MAXHOSTNAMELEN)` and then check each to see if it refers to a DNS name or only a hostname. I think there's another one in `nmap_dns.cc`.